index.js

```diff
@@ -0,0 +1,84 @@
+const express = require('express');
+const knex = require('knex');
+
+const knexConfig = require('./knexfile.js');
+
+const db = knex(knexConfig.development);
+
+const server = express();
+
+server.use(express.json());
+
+
+
+//endpoints for projects
+
+//===GETS ALL PROJECTS===
+server.get('/api/projects', (req, res) => {
+    db('projects')
+      .then(project => res.status(200).json(project))
+      .catch(err => res.status(500).json(err));
+  });
+
+
+//===GETS PROJECTS BY ID===
+  server.get('/api/projects/:id', (req, res) => {
+    const { id } = req.params;
+     db('projects')
+        .where({ id })
+        .then(project => {
+            db('actions')
+                .where({ projectId: id })
+                .then(action => {
+                    return res.status(200).json({ ...project, actions: action });
+                });
+        })
+        .catch(() => {
+            return res.status(500).json({ Error: "Project info could not be retrieved." })
+        });
+
+    });
+
+
+//===ADDS NEW PROJECT===
+server.post('/api/projects', (req, res) => {
+    const project = req.body;
+
+    db('projects')
+        .insert(project)
+        .returning('id')
```

Comment:
I'm not sure if you need the .returning() here. The insert automatically returns an id array (usually of length one) which you can then use to return the inserted project. This is applies to your post to actions to.